core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -488,13 +502,74 @@ public List<String> regexpExtractAll(String value, String regex) {
       ImmutableList.Builder<String> matches = ImmutableList.builder();
       while (matcher.find()) {
         String match = matcher.group(matcher.groupCount());
-        if (match != null && !match.isEmpty()) {
+        if (match != null) {
           matches.add(match);
         }
       }
       return matches.build();
     }
 
+    /** SQL {@code REGEXP_INSTR(value, regexp)} function.
+     *  Returns 0 if there is no match or regex is empty. Returns an exception if regex is invalid.
+     *  Uses position=1, occurrence=1, occurrencePosition=0 as default values if not specified. */
+    public Integer regexpInstr(String value, String regex) {
```

Comment:
should we use `int` in favor of `Integer`? It seems that's what used in most other cases.